models/api/main_api_adapter.py

```diff
@@ -9,35 +9,72 @@
 
 
 class ModelAdapter(dl.BaseModelAdapter):
-    def __init__(self, model_entity: dl.Model):
-        if os.environ.get("NGC_API_KEY", None) is None:
-            raise ValueError(f"Missing API key: NGC_API_KEY")
-        self.api_key = os.environ.get("NGC_API_KEY", None)
-        self.max_token = model_entity.configuration.get('max_token', 1024)
-        self.temperature = model_entity.configuration.get('temperature', 0.2)
-        self.top_p = model_entity.configuration.get('top_p', 0.7)
-        self.seed = model_entity.configuration.get('seed', 0)
-        self.stream = model_entity.configuration.get('stream', True)
-
-        super().__init__(model_entity)
 
     def load(self, local_path, **kwargs):
+        if os.environ.get("NGC_API_KEY", None) is None:
+            raise ValueError(f"Missing API key")
+
+        self.api_key = os.environ.get("NGC_API_KEY", None)
+        self.max_token = self.configuration.get('max_token', 1024)
+        self.temperature = self.configuration.get('temperature', 0.2)
+        self.top_p = self.configuration.get('top_p', 0.7)
+        self.seed = self.configuration.get('seed', 0)
+        self.stream = self.configuration.get('stream', True)
         self.nim_model_name = self.configuration.get("nim_model_name")
+
         if self.nim_model_name is None:
             raise ValueError("Missing `nim_model_name` from model.configuration, cant load the model without it")
 
     def prepare_item_func(self, item: dl.Item):
-        if ('json' not in item.mimetype or
-                item.metadata.get('system', dict()).get('shebang', dict()).get('dltype') != 'prompt'):
-            raise ValueError('Only prompt items are supported')
-        buffer = json.load(item.download(save_locally=False))
-        return buffer
+        prompt_item = dl.PromptItem.from_item(item)
+        return prompt_item
+
+    @staticmethod
+    def process_chat_messages(messages):
+        return [{'role': msg['role'], 'content': msg['content'][0][msg['content'][0]['type']]}
```

Comment:
don't make complex returns, create the values and ifs seperatly and return in the end